websub-ballerina/src/websub_generic_integration_tests/tests/integration-tests/subscriber/test_custom_subscribers.bal

```diff
@@ -0,0 +1,395 @@
+    dependsOn: ["testDispatchingByHeader"]
+}
+function testDispatchingByHeaderAndPayloadKey() {
+    http:Client clientEndpoint = new ("http://localhost:23787");
+    json jsonPayload1 = {action: "created"};
+    json jsonPayload2 = {domain: "feature"};
+    http:Request req1 = new;
+    req1.addHeader(http:CONTENT_TYPE, CONTENT_TYPE_JSON);
+    req1.addHeader(CUSTOM_SUB_MOCK_HEADER, "issue");
+    req1.setJsonPayload(jsonPayload1);
+    http:Request req2 = new;
+    req2.addHeader(http:CONTENT_TYPE, CONTENT_TYPE_JSON);
+    req2.addHeader(CUSTOM_SUB_MOCK_HEADER, "pull");
+    req2.setJsonPayload(jsonPayload2);
+
+    var response1 = clientEndpoint->post("/headerAndPayload", req1);
+    HttpResponseDetails responseDetails1 = fetchHttpResponse(response1);
+    test:assertEquals(responseDetails1.statusCode, http:STATUS_ACCEPTED, msg = "Response code mismatched");
+    test:assertEquals(fetchOutput(ID_BY_HEADER_AND_PAYLOAD_ISSUE_CREATED_LOG), BY_HEADER_AND_PAYLOAD_ISSUE_CREATED_LOG);
+
+    var response2 = clientEndpoint->post("/headerAndPayload", req2);
+    HttpResponseDetails responseDetails2 = fetchHttpResponse(response2);
+    test:assertEquals(responseDetails2.statusCode, http:STATUS_ACCEPTED, msg = "Response code mismatched");
+    test:assertEquals(fetchOutput(ID_BY_HEADER_AND_PAYLOAD_FEATURE_PULL_LOG), BY_HEADER_AND_PAYLOAD_FEATURE_PULL_LOG);
+}
+
+@test:Config {
+    dependsOn: ["testDispatchingByHeaderAndPayloadKey"]
+}
+function testDispatchingByHeaderAndPayloadKeyForOnlyHeader() {
+    http:Client clientEndpoint = new ("http://localhost:23787");
+    json jsonPayload = {action: "header_only"};
+    http:Request req = new;
+    req.addHeader(http:CONTENT_TYPE, CONTENT_TYPE_JSON);
+    req.addHeader(CUSTOM_SUB_MOCK_HEADER, "headeronly");
+    req.setJsonPayload(jsonPayload);
+
+    var response = clientEndpoint->post("/headerAndPayload", req);
+    HttpResponseDetails responseDetails = fetchHttpResponse(response);
+    test:assertEquals(responseDetails.statusCode, http:STATUS_ACCEPTED, msg = "Response code mismatched");
+    test:assertEquals(fetchOutput(ID_BY_HEADER_AND_PAYLOAD_HEADER_ONLY_LOG), BY_HEADER_AND_PAYLOAD_HEADER_ONLY_LOG);
+}
+
+@test:Config {
+    dependsOn: ["testDispatchingByHeaderAndPayloadKeyForOnlyHeader"]
+}
+function testDispatchingByHeaderAndPayloadKeyForOnlyKey() {
+    http:Client clientEndpoint = new ("http://localhost:23787");
+    json jsonPayload = {action: "keyonly"};
+    http:Request req = new;
+    req.addHeader(http:CONTENT_TYPE, CONTENT_TYPE_JSON);
+    req.addHeader(CUSTOM_SUB_MOCK_HEADER, "key_only");
+    req.setJsonPayload(jsonPayload);
+
+    var response = clientEndpoint->post("/headerAndPayload", req);
+    HttpResponseDetails responseDetails = fetchHttpResponse(response);
+    test:assertEquals(responseDetails.statusCode, http:STATUS_ACCEPTED, msg = "Response code mismatched");
+    test:assertEquals(fetchOutput(ID_BY_HEADER_AND_PAYLOAD_KEY_ONLY_LOG), BY_HEADER_AND_PAYLOAD_KEY_ONLY_LOG);
+}
```

Comment:
```suggestion
}

```

Reply:
Fixed in here https://github.com/ballerina-platform/module-ballerina-websub/pull/1/commits/7defc19195f199d3899556e82f9a35eb1ebf4bd5